// @livekit/protocol/src/gen/livekit_phone_number_pb.js
// @generated by protoc-gen-es v1.10.1 with parameter "target=dts+js"
// @generated from file livekit_phone_number.proto (package livekit, syntax proto3)
/* eslint-disable */
// @ts-nocheck

import { proto3, Timestamp } from "@bufbuild/protobuf";
import { TokenPagination } from "./livekit_models_pb.js";
import { SIPDispatchRuleInfo } from "./livekit_sip_pb.js";

/**
 * Telephony cost type enumeration
 *
 * @generated from enum livekit.TelephonyCostType
 */
export const TelephonyCostType = /*@__PURE__*/ proto3.makeEnum(
  "livekit.TelephonyCostType",
  [
    {no: 0, name: "TELEPHONY_COST_TYPE_UNSPECIFIED", localName: "UNSPECIFIED"},
    {no: 1, name: "TELEPHONY_COST_TYPE_NUMBER_RENTAL", localName: "NUMBER_RENTAL"},
    {no: 2, name: "TELEPHONY_COST_TYPE_VOICE_INBOUND", localName: "VOICE_INBOUND"},
    {no: 3, name: "TELEPHONY_COST_TYPE_VOICE_OUTBOUND", localName: "VOICE_OUTBOUND"},
    {no: 4, name: "TELEPHONY_COST_TYPE_SMS_INBOUND", localName: "SMS_INBOUND"},
    {no: 5, name: "TELEPHONY_COST_TYPE_SMS_OUTBOUND", localName: "SMS_OUTBOUND"},
  ],
);

/**
 * Phone number status enumeration
 *
 * @generated from enum livekit.PhoneNumberStatus
 */
export const PhoneNumberStatus = /*@__PURE__*/ proto3.makeEnum(
  "livekit.PhoneNumberStatus",
  [
    {no: 0, name: "PHONE_NUMBER_STATUS_UNSPECIFIED", localName: "UNSPECIFIED"},
    {no: 1, name: "PHONE_NUMBER_STATUS_ACTIVE", localName: "ACTIVE"},
    {no: 2, name: "PHONE_NUMBER_STATUS_PENDING", localName: "PENDING"},
    {no: 3, name: "PHONE_NUMBER_STATUS_RELEASED", localName: "RELEASED"},
  ],
);

/**
 * ListPhoneNumberInventoryRequest - Request to list available phone numbers
 *
 * @generated from message livekit.ListPhoneNumberInventoryRequest
 */
export const ListPhoneNumberInventoryRequest = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.ListPhoneNumberInventoryRequest",
  () => [
    { no: 1, name: "country_code", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "area_code", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "limit", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
    { no: 4, name: "page_token", kind: "message", T: TokenPagination },
  ],
);

/**
 * ListPhoneNumberInventoryResponse - Response containing available phone numbers
 *
 * @generated from message livekit.ListPhoneNumberInventoryResponse
 */
export const ListPhoneNumberInventoryResponse = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.ListPhoneNumberInventoryResponse",
  () => [
    { no: 1, name: "items", kind: "message", T: PhoneNumberInventoryItem, repeated: true },
    { no: 2, name: "next_page_token", kind: "message", T: TokenPagination },
  ],
);

/**
 * PurchasePhoneNumberRequest - Request to purchase phone numbers
 *
 * @generated from message livekit.PurchasePhoneNumberRequest
 */
export const PurchasePhoneNumberRequest = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.PurchasePhoneNumberRequest",
  () => [
    { no: 1, name: "phone_numbers", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
  ],
);

/**
 * PurchasePhoneNumberResponse - Response containing the purchased phone numbers
 *
 * @generated from message livekit.PurchasePhoneNumberResponse
 */
export const PurchasePhoneNumberResponse = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.PurchasePhoneNumberResponse",
  () => [
    { no: 1, name: "phone_numbers", kind: "message", T: PurchasedPhoneNumber, repeated: true },
  ],
);

/**
 * ListPurchasedPhoneNumbersRequest - Request to list purchased phone numbers
 *
 * @generated from message livekit.ListPurchasedPhoneNumbersRequest
 */
export const ListPurchasedPhoneNumbersRequest = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.ListPurchasedPhoneNumbersRequest",
  () => [
    { no: 1, name: "limit", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
    { no: 2, name: "page_token", kind: "message", T: TokenPagination },
  ],
);

/**
 * ListPurchasedPhoneNumbersResponse - Response containing purchased phone numbers
 *
 * @generated from message livekit.ListPurchasedPhoneNumbersResponse
 */
export const ListPurchasedPhoneNumbersResponse = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.ListPurchasedPhoneNumbersResponse",
  () => [
    { no: 1, name: "items", kind: "message", T: PurchasedPhoneNumber, repeated: true },
    { no: 2, name: "next_page_token", kind: "message", T: TokenPagination },
    { no: 3, name: "total_count", kind: "scalar", T: 5 /* ScalarType.INT32 */ },
  ],
);

/**
 * ReleasePhoneNumberRequest - Request to release purchased phone numbers
 *
 * @generated from message livekit.ReleasePhoneNumberRequest
 */
export const ReleasePhoneNumberRequest = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.ReleasePhoneNumberRequest",
  () => [
    { no: 1, name: "phone_numbers", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
  ],
);

/**
 * GlobalPhoneNumber represents a phone number with standardized format
 *
 * @generated from message livekit.GlobalPhoneNumber
 */
export const GlobalPhoneNumber = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.GlobalPhoneNumber",
  () => [
    { no: 1, name: "id", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 2, name: "e164_format", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "country_code", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 4, name: "area_code", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 5, name: "number_type", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 6, name: "locality", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 7, name: "region", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 8, name: "spam_score", kind: "scalar", T: 1 /* ScalarType.DOUBLE */ },
    { no: 9, name: "created_at", kind: "message", T: Timestamp },
    { no: 10, name: "updated_at", kind: "message", T: Timestamp },
  ],
);

/**
 * TelephonyCost represents the pricing structure for a specific telephony service
 *
 * @generated from message livekit.TelephonyCost
 */
export const TelephonyCost = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.TelephonyCost",
  () => [
    { no: 1, name: "resource", kind: "enum", T: proto3.getEnumType(TelephonyCostType) },
    { no: 2, name: "currency", kind: "scalar", T: 9 /* ScalarType.STRING */ },
    { no: 3, name: "price", kind: "scalar", T: 1 /* ScalarType.DOUBLE */ },
    { no: 4, name: "billing_unit", kind: "scalar", T: 9 /* ScalarType.STRING */ },
  ],
);

/**
 * PhoneNumberInventoryItem - Represents an available phone number for purchase
 *
 * @generated from message livekit.PhoneNumberInventoryItem
 */
export const PhoneNumberInventoryItem = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.PhoneNumberInventoryItem",
  () => [
    { no: 1, name: "phone_number", kind: "message", T: GlobalPhoneNumber },
    { no: 2, name: "capabilities", kind: "scalar", T: 9 /* ScalarType.STRING */, repeated: true },
    { no: 3, name: "costs", kind: "message", T: TelephonyCost, repeated: true },
  ],
);

/**
 * PurchasedPhoneNumber - Represents a phone number owned by a LiveKit project
 *
 * @generated from message livekit.PurchasedPhoneNumber
 */
export const PurchasedPhoneNumber = /*@__PURE__*/ proto3.makeMessageType(
  "livekit.PurchasedPhoneNumber",
  () => [
    { no: 1, name: "phone_number", kind: "message", T: GlobalPhoneNumber },
    { no: 2, name: "status", kind: "enum", T: proto3.getEnumType(PhoneNumberStatus) },
    { no: 3, name: "assigned_at", kind: "message", T: Timestamp },
    { no: 4, name: "released_at", kind: "message", T: Timestamp },
    { no: 5, name: "sip_dispatch_rule", kind: "message", T: SIPDispatchRuleInfo },
  ],
);
